### ISSUE TITLE
Dynamic flyouts are not shown when animated

### ISSUE THREAD

**Issue by didster** [MahApps/MahApps.Metro#3807]
**Describe the bug**
Dynamic flyouts (those added to FlyoutsControl.Items at runtime) dont show with animations enabled (the default).  Worked fine prior to around 2.0.0-alpha0872 and I assume related to  [this](https://github.com/MahApps/MahApps.Metro/commit/576522b81af1f78e73e0c1a8ec48e7d19e2e1fa8#diff-019ea1a0512ff16f157fb0b381a1da48) commit.

Demo app exhibits same behaviour - all flyouts (which are inlined in xaml) work fine other than dynamic button which doesn't show.

**To Reproduce**
As per demo code

```
var flyout = new DynamicFlyout
                      {
                             Header = "Dynamic flyout"
                         };
            // when the flyout is closed, remove it from the hosting FlyoutsControl
            RoutedEventHandler closingFinishedHandler = null;
            closingFinishedHandler = (o, args) =>
                {
                    flyout.ClosingFinished -= closingFinishedHandler;
                    flyoutsControl.Items.Remove(flyout);
                };
            flyout.ClosingFinished += closingFinishedHandler;

            flyoutsControl.Items.Add(flyout);

            flyout.IsOpen = true;
```

Addition of `flyout.AreAnimationsEnabled = false;` results in the flyout showing as expected.

**Expected behavior**
The flyout is shown animated

**Environment(please complete the following information):**
 - MahApps.Metro version [2.0.0-alpha0879]
 - OS: Win10 18362
 - Visual Studio 2019 (16.5.2)
 - .NET Framework: Core 3.1

**Comment by punker76** [comment]
@didster Thx for using the latest alpha hot stuff and reporting this issue. It's now fixed and will be available with the final v2.0 release.

**Comment by didster** [comment]
@punker76 Pleasure, just raised another ;-).  Thanks for the vast efforts for v2 its coming on great.
